test/OpenTelemetry.AotCompatibility.Tests/OpenTelemetry.AotCompatibility.Tests.csproj

```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <Description>Test to ensure AOT compatilibity.</Description>
+    <TargetFramework>net6.0</TargetFramework>
```

Comment:
Use `net7.0` instead?

Reply:
Targeting this test project on net6.0 or net7.0 would emit the same set of warnings. 

I'm not sure what is the expected behavior of AOT compiler when analyzing a project that targets on net6.0 and the unit test of the app does a dotnet publish on a project that targets net7.0. 
Maybe @eerhardt can shed some light on this.

Anyways, I updated the TargetFramework here to be net7.0. Thanks!

Reply:
The TFM of the test project doesn't matter in this case. What matters is the TFM of the test app being published as AOT.

Reply:
thank you @vitek-karas!